books/kestrel/c/syntax/validator.lisp

```diff
@@ -1201,14 +1168,90 @@
               and the second sub-expression has type ~x2."
              (expr-fix expr)
              (type-fix type-arg1)
-             (type-fix type-arg2)))
-  :hooks (:fix))
+             (type-fix type-arg2))))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-(define valid-funcall ((expr exprp) (type-fun typep) (types-arg type-listp))
-  (declare (ignore types-arg))
-  :guard (expr-case expr :funcall)
+(define valid-prototype-args ((types-param type-listp)
+                              (args expr-listp)
+                              (types-arg type-listp)
+                              (ellipsis booleanp)
+                              (ienv ienvp))
+  :guard (equal (len types-arg) (len args))
+  :returns (erp maybe-msgp)
+  :short "Validate a function prototype parameter list against a list of
+          arguments given the argument types."
+  :long
+  (xdoc::topstring
+   (xdoc::p
+    "Given a function type which includes a prototype,
+     two conditions must hold [C17:6.5.2.2/2]:")
+   (xdoc::ol
+    (xdoc::li
+     "The number of arguments must be the same as the number of parameter types,
+      or greater than equal when an ellipsis is present.")
```

Comment:
```suggestion
      or greater than or equal when an ellipsis is present.")
```

Reply:
I'm not readily finding where, in [C17], it says that more arguments than parameters are allowed when the function type has ellipsis, even though it seems right.